### PR TITLE
feat(link): add support for HTMLElement.click()

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #4212

## Summary

Adds a `clickHandler` that forwards the `.click()` to the internal link for non user-initiated events, stops forwarded internal event from propagating, and ensures with a test.

### CODE REVIEW

**Comment by Elijbet** [comment]
A little perplexed by the test behavior: .click() results in 1 event being received each time, while page.$eval and .callMethod() receive it twice.

File: src/components/link/link.tsx [hunk 1]
```diff
@@ -138,6 +153,15 @@ export class Link implements InteractiveComponent {
   /** the rendered child element */
   private childEl: HTMLAnchorElement | HTMLSpanElement;
 
+  /**
```
Comment (jcfranco) [inline src/components/link/link.tsx:156]: This doc/comment block is not needed. The handler name already expresses this information.

File: src/components/link/link.tsx [hunk 2]
```diff
@@ -138,6 +153,15 @@ export class Link implements InteractiveComponent {
...
+   * handle clicks on child element
+   *
+   * @param event
+   */
+  private childElClickHandler = (event: PointerEvent): void => {
+    event.stopPropagation();
```
Comment (jcfranco) [inline src/components/link/link.tsx:162]: You still need to determine if the click came from the forwarded click or not. Only stop propagation if it's internal.

File: src/components/link/link.tsx [hunk 3]
```diff
@@ -117,6 +118,20 @@ export class Link implements InteractiveComponent {
...
 
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("click")
+  clickHandler(event: PointerEvent): void {
+    /* forwards the click() to the internal link for non user-initiated events */
```
Comment (jcfranco) [inline src/components/link/link.tsx:129]: Nitpick: use `//` for single-line comments

**Review by jcfranco — CHANGES_REQUESTED** [review]

**Review by jcfranco — APPROVED** [review]
💥

**Pull request merged**